docs/src/pages/[platform]/connected-components/account-settings/change-password/react.mdx

```diff
@@ -1,3 +1,359 @@
+
+<Example>
+  <ExampleCode>
+
+```ts
+interface ValidatorOptions {
+  validationMode: 'onBlur' | 'onChange' | 'onTouched';
+  validator: (value: string) => boolean;
+  message: string;
+}
+```
+
+  </ExampleCode>
+</Example>
+
+- `onBlur` validates password on every blur event
+- `onChange` validates password on every change event
+- `onTouched` validates password on first blur, and on every change event after the first blur.
+
+For example, following is a minLength validator that validates on every change:
+
+<Example>
+  <ExampleCode>
+
+```js
+const minLength = {
+  validationMode: 'onChange',
+  validator: (password) => password.length >= 4,
+  message: 'Password must have length 4 or greater',
+};
+```
+
+  </ExampleCode>
+</Example>
+
+You can pass multiple validators to ConfirmPassword to override the default validator:
+
+<Example>
+  <ExampleCode>
+
+```jsx
+import React from 'react';
+import { AccountSettings } from '@aws-amplify/ui-react';
+
+function App() {
+  const minLength = {
+    validationMode: 'onChange',
+    validator: (password) => password.length >= 4,
+    message: 'Password must have length 4 or greater',
+  };
+  
+  const handleSuccess = () => {
+    alert('password is successfully changed!')
+  }
+  
+  return (
+    <AccountSettings.ChangePassword onSuccess={handleSuccess} validators={[minLength]}/>
+  );
+}
+```
+
+  </ExampleCode>
+</Example>
+
+### Component Overrides
+
+You can provide your own UI components to override parts of `ChangePassword`. It supports the following slots:
+
+<ReactPropsTable props={OVERRIDES} />
+
+#### Reusing default components
+
+Default components are accessible through `AccountSettings.ChangePassword.COMPONENT_NAME` (e.g. `AccountSettings.ChangePassword.NewPassword`) for your convenience. This is useful if you're interested in modifying just a small part of UI instead of overriding the whole component.
+
+<Example>
+  <ExampleCode>
+
+```jsx
+function App() {
+  const components = {
+    NewPassword: (props) => (
+      <AccountSettings.ChangePassword.NewPasswordField
+        {...props} 
+        label="Custom New Password Label" 
+      />
+    ),
+  }
+
+  return (
+    <AccountSettings.ChangePassword components={components} />
+  );
+}
+```
+
+  </ExampleCode>
+</Example>
+
+<Alert variation="warning">
+Note that spreading props is necessary so that event listeners like `onChange` or html attributes like `name` are passed correctly. If you're providing your own custom components, make sure required props are passed onto your elements.
+</Alert>
+
+#### Props and Examples
+
+Here are the required props and examples for overriding each slot.
+
+<Expander type="single">
+  <ExpanderItem title="PasswordFields" value="password-fields">
+
+`ChangePassword` has three password fields that you can override: `CurrentPassword`, `NewPassword`, and `NewPassword`. They all take the same props:
```

Comment:
`CurrentPassword`, `NewPassword`, and `ConfirmPassword`

Reply:
🤦 , thanks
```suggestion
`ChangePassword` has three password fields that you can override: `CurrentPassword`, `NewPassword`, and `ConfirmPassword`. They all take the same props:
```